UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -87,6 +88,24 @@ private void Update()
         movementVector.y = verticalMovement;
         characterController.Move(movementVector * Time.deltaTime);
 
+        if (characterController.isGrounded) {
+            //Check if underneath the character is stable
+            isUnderneathStable = false; //By default, underneath is not stable
+            RaycastHit hit;
+            Ray ray = new Ray(transform.position + characterController.center, Vector3.down);
```

Comment:
Instead of calculating the hit position with raycast you can get the same in 
`private void OnControllerColliderHit(ControllerColliderHit hit)`

hit varaible contains the `hit.normal` using which you can calculate the contact angle which would be more efficient

Reply:
I am not using OnControllerColliderHit because this will be triggered each time a collider is hit.
That could be an object we want to push, a wall in front of us, an obstacle that hits the head, etc...

Reply:
ohhh that makes sense.
But what about the situations like below case
![image](https://user-images.githubusercontent.com/33123731/95007737-b3393600-0630-11eb-86be-59f0a5f82cf0.png)